### PR TITLE
Add 'aws-configure' and 'citadel' utilities

### PR DESCRIPTION
**Pull request by tduffield**
Signed-off-by: Tom Duffield <tom@chef.io>

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -7,6 +7,7 @@ branches:
 before_install:
   - export PATH="$PATH:$HOME/ci-studio-common/bin:$HOME/tools/bin"
   - ln -sf "$TRAVIS_BUILD_DIR" "$HOME/ci-studio-common"
+  - aws-configure chef-cd
```
Comment (schisamo) [inline .travis.yml]: slick!

File: bin/aws-configure [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+You can optionally specify '<PROFILE>_AWS_DEFAULT_REGION' to determine the default region for this profile.
+
+If no PROFILE is specified, ${0##*/} will defer to the value specified in AWS_PROFILE.
+DOC
+}
+
+function enforce_requirements() {
+  # Make sure the AWS CLI is installed
+  which aws >/dev/null 2>1
```
Comment (afiune) [inline bin/aws-configure]: You missed an `&`, this should be: (more info [here](http://www.tldp.org/LDP/abs/html/io-redirection.html))
```bash
which aws >/dev/null 2>&1
```

File: bin/aws-configure [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+function usage() {
+  cat /dev/stdin <<DOC
+Usage: ${0##*/} [PROFILE]
```
Comment (afiune) [inline bin/aws-configure]: So happy to see thiiiiiiss!!! 👍 💯

File: bin/citadel [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+ENVIRONMENT VARIABLES:
+    CITADEL_BUCKET      The name of the S3 bucket where citadel files are kept. (default: chef-cd-citadel)
+    CITADEL_PROFILE     The name of the AWS CLI profile with access to citadel. (default: chef-cd)
+DOC
+}
+
+function enforce_requirements() {
+  # Make sure the AWS CLI is installed
+  which aws >/dev/null 2>1
```
Comment (afiune) [inline bin/citadel]: Same here, missed an `&`

File: bin/citadel [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+function usage() {
+  cat /dev/stdin <<DOC
+Usage: ${0##*/} FILE
+
+A Bash utility that prints the contents of the given FILE to STDOUT.
```
Comment (afiune) [inline bin/citadel]: I think I was confused with the usage of this command, "Given File"? I think we should say that the file lives in an S3 bucket. But maybe that was just me. Not sure the use case for this command though so totally a nit.

File: tools/aws.sh [hunk 6]
```diff
@@ -17,4 +17,4 @@
 #
 
 pip install --user awscli
-ln -sf $HOME/.local/bin/aws $HOME/tools/bin/aws
+ln -sf $HOME/.local/bin/aws $HOME/tools/bin/aws
```
Comment (afiune) [inline tools/aws.sh]: missing extra line?

Comment (afiune) [reply tools/aws.sh]: Also what if the link already exists?
```
➜  ln -s /tmp/l /tmp/b
ln: /tmp/b: File exists
```

Comment (tduffield) [reply tools/aws.sh]: The `-f` will force it.

**Pull request merged**